### ISSUE TITLE
RFE: tmt command  aliases.

### ISSUE THREAD

**Issue by pvalena** [psss/tmt#414]
Actually what I expected would be for tmt to know such aliases. 

Where you recommend `alias reserve=...` to create command `reserve` I'd expect to have `$ tmt reserve` built in, or `$ tmt start` etc.. (So it's all fine, except I'd like tmt to actually implement it, not just recommend.)

_Originally posted by @pvalena in https://github.com/psss/tmt/pull/404#issuecomment-707632170_

**Comment by pvalena** [comment]
Also please take a look at my useful tmt commands I've noted:
https://gist.github.com/pvalena/b4a94178c13e861fa8520c84aa8319b8

**Comment by jkonecny12** [comment]
Hi, thanks for all your help.

I would like to add alias which is combining existing ``start`` and ``retest`` aliases.
```
	tmt run -vvv --id $(TEST_TMT_ID) --until report discover -f execute -f
```

This will run everything until ``report`` step and if they were already executed it will skip then and run just ``discover`` and ``execute`` steps again. That's pretty convenient for local testing and makes the current aliases a bit easier to use.

**Comment by pemensik** [comment]
I think just aliases are not way to go, because they would be hard to document and create useful --help for them. I think some tmt subcommands with at least minimal help documentation should be provided.

**Comment by pvalena** [comment]
> I think just aliases are not way to go, because they would be hard to document and create useful --help for them. I think some tmt subcommands with at least minimal help documentation should be provided.

Every tmt sub-option has its own customised help (based on plugins used as well). E.g.:

```log
$ tmt run provision --how virtual --help
Usage: tmt run provision [OPTIONS]

  Local virtual machine using testcloud

  Minimal config which uses the latest fedora image:

      provision:
          how: virtual

  Here's a full config example:

      provision:
          how: virtual
          image: fedora
          user: root
          memory: 2048
[ . . . ]
```

vs

```log
$ tmt run provision --how container --help
Usage: tmt run provision [OPTIONS]

  Create a new container using podman

  Example config:

      provision:
          how: container
          image: fedora:latest

  In order to always pull the fresh container image use 'pull: true'.

  In order to run the container with different user as the default 'root',
  use 'user: USER'.
[ . . . ]
```

You can also list various methods / plugins that the step supports (not just view their help):
```log
 tmt run execute --help
Usage: tmt run execute [OPTIONS]

  Run tests using the specified executor.

  Supported methods (tmt by default):

  tmt .................. Use the internal tmt executor to execute tests
  upgrade .............. Perform system upgrade during testing.

  Use 'tmt run execute --help --how <method>' to learn more about given
  execute method and all its supported options.

[ . . . ]
```

_ _ _ _ 
Can you give some examples of `subcommands` which lack documentation?

**Comment by pemensik** [comment]
What I meant is lacking any documentation are aliases provided as part of #285, [examples#aliases](https://tmt.readthedocs.io/en/latest/examples.html#aliases). Those have not small amount of multiple parameters, but nothing is describing what one alias is used for. If I would have 10 such aliases, I would need some help system.

yes, I would expect to have some place, where I can look available aliases and their documentation. That is why I think those should be integrated into tmt somehow, just like you requested, so it would offer similar description that tmt subcommands always have. Or at least file defined actions, which can contain also brief description presented by built-in help system.

**Comment by pvalena** [comment]
I completely agree. Integration of those with tmt is a key for further tmt adoption.

**Comment by psss** [comment]
I believe the main use case has been covered by the `tmt try` command. Closing, feel free to reopen if it does not seem right.

**Comment by pvalena** [comment]
It's something different, kind-of implements some things... To clarify: `try` shares one machine (same root ID?), right? If I run multiple IDs (f.e. per-repository or project) I'd always have to specify that on a commandline.

In any case; shells & other tools support aliases allright :) so it was more for a beginner friendly use-case. Turns out (AFAIR) people usually have a very specific usecase in mind when starting with tmt, and need to dive in to tmt configs instead. Therefore feel free to keep it closed.
